.cirrus.yml

```diff
@@ -448,6 +446,33 @@ osx_alt_build_task:
     always: *runner_stats
 
 
+# Build freebsd release natively on a FreeBSD VM.
+freebsd_alt_build_task:
+    name: "FreeBSD Cross"
+    alias: freebsd_alt_build
+    # Docs: ./contrib/cirrus/CIModes.md
+    only_if: *no_multiarch_rhel
+    depends_on:
+        - build
+    env:
+        <<: *stdenvars
+        # Functional FreeBSD builds must be built natively since they depend on CGO
+        CIRRUS_SHELL: "/bin/sh"
+        TEST_FLAVOR: "altbuild"
+        ALT_NAME: 'FreeBSD Cross'
+    freebsd_instance:
+        image_family: freebsd-13-2
+    setup_script:
+        - pkg install -y gpgme bash go-md2man gmake gsed gnugrep go pkgconf
```

Comment:
Installing stuff at runtime is prone to introducing flakes, esp. on long-term release branches (even if non-rhel).  OTOH, adding freebsd into our CI VM image build pipeline is probably no small feat.  Certainly not one I would wish on anybody not intimately familiar with it.

However, since this is only a "does it compile" check, maybe a compromise could be to restrict the scope of the job further.  Consider something like (untested):

```yaml
only_if:
    $CIRRUS_CRON != 'multiarch' &&
    ($CIRRUS_BRANCH == 'main' || $CIRRUS_BASE_BRANCH == 'main')
```

So it only ever runs on `main` or PRs targeted at `main`.

Reply:
I would like to get to a point where we have a FreeBSD CI VM image - I had a look at the code to build them yesterday and got quite confused. Long term, I would like to be able to run system tests in CI, perhaps  non-blocking. For now, I restricted this task as you suggest.

Reply:
> I had a look at the code to build them yesterday and got quite confused

Oh yes, it's definitely not simple.  Assuming ssh and the google-cloud services work similar to Linux, I'm pretty sure the tooling in c/automation_images should be fine.  I'll add this to my TODOs to investigate further.  It would also help with potentially getting buildah and skopeo testing added as well.